addons/web_editor/static/src/js/editor/odoo-editor/src/commands/commands.js

```diff
@@ -706,11 +707,40 @@ export const editorCommands = {
             return selectedNodes.flatMap(node => {
                 let font = closestElement(node, 'font') || closestElement(node, 'span');
                 const children = font && descendants(font);
-                if (font && (font.nodeName === 'FONT' || (font.nodeName === 'SPAN' && font.style[mode]))) {
+                const closestGradient = closestElement(node, '[style*="background-image"]');
+                const isGradient = font && isColorGradient(font.style["background-image"]);
+                const isTextGradient = isGradient && font.classList.contains("text-gradient");
+                if (
+                    font &&
+                    (font.nodeName === "FONT" || (font.nodeName === "SPAN" && font.style[mode])) &&
+                    (isColorGradient(color) || !isGradient)
+                ) {
                     // Partially selected <font>: split it.
                     const selectedChildren = children.filter(child => selectedNodes.includes(child));
                     if (selectedChildren.length) {
-                        font = splitAroundUntil(selectedChildren, font);
+                        const isGradientBeingUpdated = closestGradient && isColorGradient(color);
+                        const splitnode = isGradientBeingUpdated ? closestGradient : font;
+                        font = splitAroundUntil(selectedChildren, splitnode);
+                        if (isGradientBeingUpdated) {
+                            const classRegex = mode === 'color' ?  TEXT_CLASSES_REGEX : BG_CLASSES_REGEX;
+                            // When updating a gradient, remove color applied to its descendants.
+                            // This ensures the gradient remains visible without being overwritten by a descendant's color.
```

Comment:
This comment line looks to be too long on my screen on Github. Check it in your editor. If it's too long, wrap it correctly. If it's just my screen on github then it's ok.